core/orch_test.go

```diff
@@ -974,8 +977,22 @@ func TestProcessPayment_GivenReceiveTicketError_ReturnsError(t *testing.T) {
 
 	time.Sleep(time.Millisecond * 20)
 	assert := assert.New(t)
-	assert.EqualError(err, "ReceiveTicket error")
+	assert.EqualError(err, errInvalidTicketSignature.Error())
 	recipient.AssertNumberOfCalls(t, "RedeemWinningTicket", 1)
+
+	// Does not loop through tickets if won==false and error is a fatal receive error
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", false, pm.NewFatalReceiveErr(errors.New("ReceiveTicket error"))).Once()
+	err = orch.ProcessPayment(*defaultPaymentWithTickets(t, senderParams), manifestID)
+	_, ok := err.(*pm.FatalReceiveErr)
+	assert.True(ok)
+
+	// Redeem winning tickets if won==true and not a signature error (but err != nil)
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", true, errors.New("ReceiveTicket error")).Once()
+	recipient.On("ReceiveTicket", mock.Anything, mock.Anything, mock.Anything).Return("", true, nil).Once()
+	recipient.On("RedeemWinningTicket", mock.Anything, mock.Anything, mock.Anything).Return(nil).Twice()
+	err = orch.ProcessPayment(*defaultPaymentWithTickets(t, senderParams), manifestID)
+	assert.EqualError(err, "ReceiveTicket error")
+	recipient.AssertNumberOfCalls(t, "RedeemWinningTicket", 2)
```

Comment:
We should add a sleep (i.e. `time.Sleep(time.Millisecond * 20)`) before this line or else this asertion could fail sometimes if it is executed before `RedeemWinningTicket()` is called in the goroutine started by `ProcessPayment()`.